weaviate/connection/rest.go

```diff
@@ -53,6 +54,39 @@ func NewConnection(scheme string, host string, httpClient *http.Client, headers
 	return connection
 }
 
+// WaitForWeaviate waits until weaviate is started up and ready
+func (con *Connection) WaitForWeaviate(startup_period int) error {
+	if startup_period < 0 {
+		return errors.New("'startup_period' needs to be an integer larger than zero")
+	}
+	ticker := time.NewTicker(time.Second)
+	start_time := time.Now()
+	for {
+		t := <-ticker.C
+		ctx, cancelFunc := context.WithTimeout(context.Background(), time.Second)
+		response, err := con.RunREST(ctx, "/.well-known/ready", http.MethodGet, nil)
+		var isReady bool
+		switch {
+		case err != nil:
+			isReady = false
+		case response.StatusCode == 200:
+			isReady = true
+		default:
+			isReady = false
+
+		}
+
+		cancelFunc()
+		if isReady {
+			return nil
+		}
+		if t.After(start_time.Add(time.Duration(startup_period) * time.Second)) {
+			return fmt.Errorf("weaviate did not start up in %d seconds. Either the Weaviate URL %s is wrong or Weaviate did not start up in the interval given in 'startup_period'", startup_period, con.basePath)
+		}
+		fmt.Printf("Weaviate not yet up waiting another second. Current time: %v\n", t)
```

Comment:
Probably a `log.Printf` would be better here, as a formatted timestamp would already be included. 

Also, this sentence is a bit hard to understand at first glance:
```
Weaviate not yet up waiting another second
```